static/app/views/issueList/searchBar.spec.tsx

```diff
@@ -0,0 +1,303 @@
+      // This should never get called
+      const loader = jest.fn(x => x);
+
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} tagValueLoader={loader} />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByRole('textbox'));
+      userEvent.keyboard('timesSeen:');
+
+      expect(loader).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('Recent Searches', function () {
+    it('saves search query as a recent search', function () {
+      const saveRecentSearch = MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/recent-searches/',
+        method: 'POST',
+        body: {},
+      });
+      const loader = jest.fn();
+      const onSearch = jest.fn();
+
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar
+            {...defaultProps}
+            tagValueLoader={loader}
+            onSearch={onSearch}
+          />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByRole('textbox'));
+      userEvent.keyboard('url:"fu"');
+
+      expect(loader).toHaveBeenCalledWith('url', 'fu');
+
+      expect(screen.getByTestId('smart-search-dropdown')).toBeInTheDocument();
+      expect(screen.queryByRole('listitem')).not.toBeInTheDocument();
+
+      userEvent.keyboard('{Enter}');
+      expect(onSearch).toHaveBeenCalledWith('url:"fu"');
+
+      expect(saveRecentSearch).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.objectContaining({
+          data: {
+            query: 'url:"fu"',
+            type: 0,
+          },
+        })
+      );
+    });
+
+    it('queries for recent searches', function () {
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByRole('textbox'));
+      userEvent.keyboard('is:');
+
+      expect(recentSearchMock).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.objectContaining({
+          query: {
+            query: 'is:',
+            limit: 3,
+            type: 0,
+          },
+        })
+      );
+    });
+
+    it('cycles through keyboard navigation for selection', async function () {
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      const textarea = screen.getByRole('textbox');
+
+      // Keyboard navigate to first item and select
+      userEvent.click(textarea);
+      userEvent.keyboard('t');
+      await waitFor(() =>
+        expect(screen.getAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument()
+      );
```

Comment:
```suggestion
      expect(await screen.findAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument();
```

Reply:
Does this work?

Reply:
Nope :(

Reply:
booooo

Reply:
For some reason it takes some time for the items to pop up

Reply:
yeah it's fine. Let's just get stuff into RTL and we can worry about performance cleanup later